tracing-subscriber/src/registry/fmt.rs

```diff
@@ -0,0 +1,315 @@
+use crate::{
+    field::RecordFields,
+    fmt::{format, FormatEvent, FormatFields, MakeWriter},
+    layer::{Context, Layer},
+    registry::{LookupMetadata, LookupSpan, Registry, SpanData},
+};
+use smallvec::{smallvec, SmallVec};
+use std::{cell::RefCell, fmt, io, marker::PhantomData};
+use tracing_core::{
+    span::{Attributes, Id, Record},
+    Event, Subscriber,
+};
+
+/// A `Subscriber` that logs formatted representations of `tracing` events.
+pub struct FmtLayer<
+    S = Registry,
+    N = format::DefaultFields,
+    E = format::Format<format::Full>,
+    W = fn() -> io::Stdout,
+> {
+    // TODO(david): don't force boxing here. consider:
+    // - starting on https://github.com/tokio-rs/tracing/issues/302
+    // - making it a generic param; defaulting to a boxed impl.
+    // - rename this, because this isn't per-subscriber filtering.
+    is_interested: Box<dyn Fn(&Event<'_>) -> bool + Send + Sync + 'static>,
+    make_writer: W,
+    fmt_fields: N,
+    fmt_event: E,
+    _inner: PhantomData<S>,
+}
+
+impl<S, N, E, W> fmt::Debug for FmtLayer<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("FmtLayer").finish()
```

Comment:
TIOLI: we _can_ format `W`, `N`, and `E` here if they have debug impls...it may not be _that_ useful, but maybe we should anyway...